packages/eslint-plugin-next-on-pages/src/rules/no-unsupported-configs.ts

```diff
@@ -6,44 +6,64 @@ import { extractPaths } from '../utils/extract-paths';
 // Note: the rule now only checks for property name, it will probably need to also include case in which we do accept a property but not
 //       certain values for it
 
-/** configs that next-on-pages does support */
-const supportedConfigs: Readonly<Set<string>> = new Set([
-	'env',
-	'basePath',
-	'rewrites',
-	'redirects',
-	'headers',
-	'pageExtensions',
-	'webpack',
-	'generateBuildId',
-	'eslint',
-	'typescript',
-	'experimental/appDir',
-]);
-
-/** configs that next-on-pages does not support and likely never will */
-const indefinitelyUnsupportedConfigs: Readonly<Set<string>> = new Set([
-	'compress',
-	'serverRuntimeConfig',
-	'publicRuntimeConfig',
-	'httpAgentOptions',
-	'distDir',
-	'onDemandEntries',
-	'exportPathMap',
-	'devIndicators',
-]);
-
-/** configs that next-on-pages does not support right now but likely will */
-const currentlyUnsupportedConfigs: Readonly<Set<string>> = new Set([
-	'assetPrefix',
-	'images',
-	'poweredByHeader',
-	'generateEtags',
-	'trailingSlash',
-	'reactStrictMode',
-	'i18n',
-	'experimental/turbo',
-]);
+type Config = { name: string; support: '✅' | 'N/A' | '❌' | '🔄' };
+
+// configs taken from https://github.com/cloudflare/next-on-pages/blob/main/packages/next-on-pages/docs/supported.md#nextconfigjs-properties
+// NOTE: to some we need to add the `experimental/` prefix and the Runtime Config is split into two
+const configs: Config[] = [
```

Comment:
The use of emojis here is a bit unconventional 😅, but I thought of doing this to keep the correlation between this and the [next.config.js props table](https://github.com/cloudflare/next-on-pages/blob/main/packages/next-on-pages/docs/supported.md#nextconfigjs-properties) as clear as possible (so that it's then easier to keep them in sync)

That's also why I changes the creation of the three config sets to all be derived from this list, so that we can have a single list to compare with the docs table (having manually typed support sets made making sure that the rule was in sync with our docs much much much harder).